frb_codegen/src/ir/func.rs

```diff
@@ -5,6 +5,7 @@ pub struct IrFunc {
     pub name: String,
     pub inputs: Vec<IrField>,
     pub output: IrType,
+    pub error_output: Option<IrType>,
     pub fallible: bool,
```

Comment:
Not sure whether `error_output` and `fallible` kind of duplicated? Or shall we use an enum to represent all states?

e.g.

```
enum {
  CustomError(IrType),
  AnyhowError,
  Infallible,
}
```

or

```
enum {
  CustomError(IrType),
  Infallible,
}
```

etc